src/components/input/math-input.js

```diff
@@ -255,36 +255,81 @@ const MathInput = React.createClass({
         });
     },
 
+    _forwardGlobalKeydown(e) {
+        if (e.keyCode === 13 || e.keyCode === 27) { // Enter, Esc
+            this.props.keypadElement.dismiss();
+            return;
+        }
+        if (e.fakeForMathquill) {
+            return;
+        }
+        const e2 = new KeyboardEvent(e.type, e);
+        Object.defineProperty(e2, 'keyCode', {get: () => e.keyCode});
+        Object.defineProperty(e2, 'charCode', {get: () => e.charCode});
+        Object.defineProperty(e2, 'which', {get: () => e.which});
+        e2.fakeForMathquill = true;
+        this.mathField.fakeTextarea.value = '';
+        this.mathField.fakeTextarea.dispatchEvent(e2);
+        this._postKeyEvent();
+    },
+
+    _forwardGlobalKeypress(e) {
+        if (e.fakeForMathquill) {
+            return;
+        }
+        // Don't allow spaces, since they invalidate a lot of expressions in our grader
+        if (e.charCode === 32) {
+            return;
+        }
+        const e2 = new KeyboardEvent(e.type, e);
+        Object.defineProperty(e2, 'keyCode', {get: () => e.keyCode});
+        Object.defineProperty(e2, 'charCode', {get: () => e.charCode});
+        Object.defineProperty(e2, 'which', {get: () => e.which});
+        e2.fakeForMathquill = true;
+        this.mathField.fakeTextarea.value = String.fromCharCode(e.charCode);
```

Comment:
On initial read I'd expect this to continuously set the input to the value of the last-pressed key. Is that incorrect? :D